### PR TITLE
Fix compile issue, add geo module as dependency

### PR DESCRIPTION
**Pull request by penghuo**
Signed-off-by: penghuo <penghuo@gmail.com>

### Description
Fix build issue with 2.3 release. OpenSearch core move geo to separate module. https://github.com/opensearch-project/OpenSearch/pull/4180. Plugin need add geo module as dependency.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: legacy/build.gradle [hunk 1]
```diff
@@ -92,6 +92,8 @@ dependencies {
     implementation group: 'org.json', name: 'json', version:'20180813'
     implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.10'
     implementation group: 'org.opensearch', name: 'opensearch', version: "${opensearch_version}"
+    // add geo module as dependency. https://github.com/opensearch-project/OpenSearch/pull/4180/.
```
Comment (saratvemulapalli) [inline legacy/build.gradle:95]: implementation will result into a jar hell. Because OpenSearch will include this jar.

Comment (penghuo) [reply legacy/build.gradle:95]: Done.

**Review by saratvemulapalli — DISMISSED** [review]
Had a chat with @penghuo. 
Looks like the plugin fails to find the class during runtime. 
We'll do a follow up as including another geo.jar doesnt hurt and will take care of it for next release.

**Comment by penghuo** [comment]
> Had a chat with @penghuo. Looks like the plugin fails to find the class during runtime. We'll do a follow up as including another geo.jar doesnt hurt and will take care of it for next release.

Create Issue. https://github.com/opensearch-project/OpenSearch/issues/4461

**Review by saratvemulapalli — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]

**Pull request merged**